miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/IndexDistanceToolController.java

```diff
@@ -87,20 +134,50 @@ public ModelAndView getTool(ModelMap model) {
   }
 
   @RequestMapping(value = "/rest/indexdistance", method = RequestMethod.POST)
-  public @ResponseBody Set<IndexDistanceWarningDto> checkIndices(@RequestBody IndexDistanceRequestDto requestObject) {
-    List<String> indices = requestObject.getIndices();
+  public @ResponseBody IndexDistanceResponseDto checkIndices(@RequestBody IndexDistanceRequestDto requestObject) {
+    List<String> indices = requestObject.getIndices().stream()
+        .map(String::trim)
+        .map(line -> line.replaceAll("\\W+", "-")) // remove any spaces, commas, dashes, etc. used to separate dual index sequences
+        .collect(Collectors.toList());
+    int shortestIndexLength = getShortestIndexLength(
+        indices.stream().map(index -> index.contains("-") ? index.split("-")[0] : index).collect(Collectors.toList())); // compare against
+                                                                                                                        // first index
     Set<IndexDistanceWarningDto> results = new HashSet<>();
 
     for (int i = 0; i < indices.size(); i++) {
       for (int j = i+1; j < indices.size(); j++) {
-        int editDistance = Index.checkEditDistance(indices.get(i), indices.get(j));
+        int editDistance = Index.checkEditDistance(truncate(indices.get(i), shortestIndexLength),
+            truncate(indices.get(j), shortestIndexLength));
         if (editDistance < requestObject.getMinimumDistance()) {
           results.add(new IndexDistanceWarningDto(indices.get(i), indices.get(j), editDistance));
         }
       }
     }
 
-    return results;
+    IndexDistanceResponseDto response = new IndexDistanceResponseDto(results, shortestIndexLength);
+    return response;
+  }
+
+  public static int getShortestIndexLength(List<String> indices) {
+    String initialValue = "REALLY_LONG_VALUE_THAT_WILL_SURELY_GET_REPLACED_BUT_HERE_ARE_SOME_MORE_CHARACTERS_JUST_IN_CASE";
+    String shortestIndex = indices.stream()
```

Comment:
You can do `indicies.stream().filter(Objects::nonNull).mapToInt(String::length).min().getAsInt()`.